cli/src/semdep/parsers/poetry.py

```diff
@@ -0,0 +1,67 @@
+from pathlib import Path
+from typing import List
+from typing import Optional
+
+from parsy import string
+from parsy import success
+
+from semdep.parsers.util import mark_line
+from semdep.parsers.util import not_any
+from semdep.parsers.util import safe_path_parse
+from semdep.parsers.util import transitivity
+from semgrep.semgrep_interfaces.semgrep_output_v1 import Ecosystem
+from semgrep.semgrep_interfaces.semgrep_output_v1 import FoundDependency
+from semgrep.semgrep_interfaces.semgrep_output_v1 import Pypi
+
+value = not_any(["\n", "["]).map(lambda x: x.strip('"')) | (
+    string("[") >> not_any(["]"]) << string("]")
+).map(lambda _: "")
+
+key_value = not_any([" ", "\n"]).bind(
+    lambda key: string(" = ") >> value.bind(lambda value: success((key, value)))
+)
+
+poetry_dep = mark_line(
+    string("[[package]]\n") >> key_value.sep_by(string("\n")).map(lambda x: dict(x))
+)
+
+poetry_sub_dep = (string("[") >> not_any(["]"]) << string("]\n")) >> key_value.sep_by(
+    string("\n")
+).map(lambda _: None)
+
+
+poetry = (poetry_dep | poetry_sub_dep).sep_by(string("\n\n")).map(
```

Comment:
Might be helpful to add comments above each grammar rule with some examples of what you expect to parse?